raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -478,6 +472,15 @@ contract TokenNetwork is Utils {
         participant1_amount = max(participant1_amount - participant1_locked_amount, 0);
         participant2_amount = max(participant2_amount - participant2_locked_amount, 0);
 
+        assert(participant1_amount <= total_deposit);
```

Comment:
Until this morning I would have left those as asserts as it "should not happen". But I had a talk with Christian today and showed him these lines. 

He said these should still be `require()`. Overflow here can happen if the appropriate input is given to this function. The only way this should be an assert is if we actually somehow in the code path that leads here have already checked that no overflow can happen. Iff we already did then this should be an assert here. If not a require.

The thinking is that there will be automatic tools that could be testing the program for correctness, feeding it with random input. Or trying to verify correctness through formal proofing tools. Asserts should never ever be reached under any input or conditions.